packages/eds-core-react/src/components/Datepicker/props.ts

```diff
@@ -0,0 +1,108 @@
+import { MutableRefObject, ReactNode } from 'react'
+import { CalendarState, RangeCalendarState } from '@react-stately/calendar'
+import { Variants } from '../types'
+import { HelperTextProps } from '../InputWrapper/HelperText'
+
+type DateRange = { from: Date | null; to: Date | null }
+
+export type HeaderProps<T = CalendarState | RangeCalendarState> = {
+  setYear: (year: number) => void
+  setMonth: (month: number) => void
+  year: number
+  month: number
+  state: T
+}
+
+export type DatePickerProps = Partial<{
+  /**
+   * The variant / state of the datepicker field
+   * @default undefined
+   */
+  variant: Variants
+  /**
+   * Whether the datepicker field is disabled
+   * @default false
+   */
+  disabled?: boolean
+  // eslint-disable-next-line react/no-unused-prop-types
+  readOnly?: boolean
+  /**
+   * Props to set the helper text
+   */
+  helperProps?: HelperTextProps
+  /**
+   * Controlled value
+   */
+  value: Date | null
+  /**
+   * The function to call when the datepicker field value changes
+   */
+  onChange: (date: Date | null) => void
+  /**
+   * The field label
+   */
+  label: string
+  /**
+   * minValue sets a limit to how far back the datepicker allows you to go
+   * note: this does not prevent you from typing in a date that is before the minValue
+   * @default undefined
+   */
+  minValue: Date
+  /**
+   * maxValue sets a limit to how far back the datepicker allows you to go
+   * note: this does not prevent you from typing in a date that is after the maxValue
+   * @default undefined
+   */
+  maxValue: Date
+  /**
+   * isDateUnavailable is used to disable specific dates (e.g. weekends)
+   * @param v
+   */
+  isDateUnavailable?: (v: Date) => boolean
```

Comment:
I had a quick look at the props in  `react-area` and saw that they are using `isDateUnavailable`. Because of this, I think it makes sense to keep it consistent.

Reply:
Yeah, I kept the parameters matching as best I could to easily track stuff between our code and react aria's